src/awkward/highlevel.py

```diff
@@ -1365,12 +1429,18 @@ def show(
         valuestr = prettyprint_valuestr(
             self, limit_rows, limit_cols, formatter=formatter_impl
         )
+
+        out = ""
         if type:
             tmp = io.StringIO()
             self.type.show(stream=tmp)
-            out = "type: " + tmp.getvalue() + valuestr
-        else:
-            out = valuestr
+            out += "type: " + tmp.getvalue()
+        if axes:
+            if self.named_axis:
+                out += f"axes: {_prettify_named_axes(self.named_axis)}\n"
+            else:
+                out += "axes: " + ", ".join(map(str, self.positional_axis)) + "\n"
+        out += valuestr
```

Comment:
Prefer lifting the `StringIO` object here to avoid allocating small strings:
```python
out_io = io.StringIO()
if type:
    out_io.write("type: ")
    self.type.show(stream=out_io)
if axes and self.named_axis:
    out_io.write("axes: ")
    out_io.write(_prettify_named_axes(self.named_axis))
    out_io.write("\n")
out_io.write(valuestr)    
```

It won't be a big performance change either way.